### PR TITLE
Rename Dataplex transport to GcpLineage

### PR DESCRIPTION
**Pull request by ddebowczyk92**
Rename Dataplex transport to GcpLineage

> **Note:** All schema changes require discussion. Please [link the issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) for context.

- [ ] Your change modifies the [core](https://github.com/OpenLineage/OpenLineage/blob/main/spec/OpenLineage.json) OpenLineage model
- [ ] Your change modifies one or more OpenLineage [facets](https://github.com/OpenLineage/OpenLineage/tree/main/spec/facets)

If you're contributing a new integration, please specify the scope of the integration and how/where it has been tested (e.g., Apache Spark integration supports `S3` and `GCS` filesystem operations, tested with AWS EMR).

### Checklist

- [x] You've [signed-off](https://github.com/OpenLineage/OpenLineage/blob/main/why-the-dco.md) your work
- [x] Your pull request title follows our [guidelines](https://github.com/OpenLineage/OpenLineage/blob/main/CONTRIBUTING.md#creating-pull-requests)
- [ ] Your changes are accompanied by tests (_if relevant_)
- [x] Your change contains a [small diff](https://kurtisnusbaum.medium.com/stacked-diffs-keeping-phabricator-diffs-small-d9964f4dcfa6) and is self-contained
- [x] You've updated any relevant documentation (_if relevant_)
- [ ] Your comment includes a one-liner for the changelog about the specific purpose of the change (_not required for changes to tests, docs, or CI config_)
- [ ] You've versioned the core OpenLineage model or facets according to [SchemaVer](https://docs.snowplowanalytics.com/docs/pipeline-components-and-applications/iglu/common-architecture/schemaver) (_if relevant_)
- [ ] You've added a [header](https://github.com/OpenLineage/OpenLineage/tree/main/.github/header_templates.md) to source files (_if relevant_)

----
SPDX-License-Identifier: Apache-2.0\
Copyright 2018-2024 contributors to the OpenLineage project

### CODE REVIEW

**Comment by mobuchowski** [comment]
Code formatting fails @ddebowczyk92

File: website/docs/client/java/partials/java_transport.md [hunk 1]
```diff
@@ -774,20 +774,20 @@ openlineage.transport.credentialsFile=path/to/credentials.json
...
 
 
-DataplexConfig dataplexConfig = new DataplexConfig();
+GcpLineageTransportConfig gcplineageConfig = new GcpLineageTransportConfig();
 
-dataplexConfig.setProjectId("your_kinesis_stream_name");
-dataplexConfig.setLocation("your_aws_region");
-dataplexConfig.setMode("sync");
-dataplexConfig.setCredentialsFile("path/to/credentials.json");
+gcplineageConfig.setProjectId("your_kinesis_stream_name");
```
Comment (julienledem) [inline website/docs/client/java/partials/java_transport.md:783]: do you want to change the project id example here?

**Comment by pawel-big-lebowski** [comment]
How about `dataplex` java client config entries and their docs? Shall it remain as is?

File: website/docs/client/java/partials/java_transport.md (hunk 1 above)
Comment (ddebowczyk92) [reply website/docs/client/java/partials/java_transport.md]: done

**Comment by ddebowczyk92** [comment]
> How about `dataplex` java client config entries and their docs? Shall it remain as is?

I changed it as well

**Review by mobuchowski — APPROVED** [review]

**Pull request merged**